### PR TITLE
📝 how to configure a new study

### PR DESCRIPTION
**Pull request by fiendish**
Addresses #121 and #108

### CODE REVIEW

File: docs/source/tutorial/study_configuration.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        short_name: 'Cat Study'
+        name: 'The long study of cats'
+        authority: 'dbgap'
+        version: 'v2.p1'
+        release_status: 'Pending'
+        category: 'Cancer'
+        attribution: 'https://www.meowmeowmeow.cat/study/attribution.html'
+
+    investigators:
+        - 'IG_12345678'
```
Comment (parimalak) [inline docs/source/tutorial/study_configuration.rst]: do we assume investigator is already in dataservice?

Comment (fiendish) [reply docs/source/tutorial/study_configuration.rst]: I think we should start requiring it.

Comment (parimalak) [reply docs/source/tutorial/study_configuration.rst]: so, the  study registration  should also include investigator registration.

Comment (fiendish) [reply docs/source/tutorial/study_configuration.rst]: I think the investigator should be registered completely separately from this process.

File: docs/source/tutorial/extract.rst [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+The operations list
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    operations = [
+        ...
+    ]
+
+This is a list of special functions that implement all of the
```
Comment (znatty22) [inline docs/source/tutorial/extract.rst]:
```suggestion
This is a list of special functions that the extract stage will execute to select subsets of source data, clean, and map the data to the desired attributes and values. The most useful functions are already written for you. You just have to invoke them appropriately.  
```

File: docs/source/tutorial/extract.rst [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+    value_map(
+        in_col="participant",
+        m={
+            r"PID(\d+)": lambda x: int(x),  # strip PID and 0-padding
+        },
+        out_col=CONCEPT.PARTICIPANT.ID
+    )
+
+This says "Use the ``participant`` column as input (in_col=), replace
```
Comment (znatty22) [inline docs/source/tutorial/extract.rst]:
```suggestion
This says "Use the ``participant`` column as input (`in_col=`), replace
```

File: docs/source/tutorial/extract.rst [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+
+    operations = [
+        value_map(
+            in_col="participant",
+            m={
+                r"PID(\d+)": lambda x: int(x),  # strip PID and 0-padding
+            },
+            out_col=CONCEPT.PARTICIPANT.ID
+        ),
+        value_map(
```
Comment (znatty22) [inline docs/source/tutorial/extract.rst]: Lets change this to use `keep_map`

File: docs/source/tutorial/extract.rst [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+            in_col="mother",
+            m=lambda x: x,
+            out_col=CONCEPT.PARTICIPANT.MOTHER_ID
+        ),
+        value_map(
```
Comment (znatty22) [inline docs/source/tutorial/extract.rst]: Lets change this to use `keep_map`

File: docs/source/tutorial/extract.rst [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+This says "Use the ``participant`` column as input (in_col=), replace
+everything that matches the regular expression pattern ``^PID(\d+)$`` with just
+the captured part and remove the zero padding by running the captured part
+through the function ``lambda x: int(x)`` (m={...}), and then output the result
+to a ``CONCEPT.PARTICIPANT.ID`` column (out_col=)."
+
+The resulting intermediate output will look like:
+
+.. csv-table::
+    :header: "index", "<CONCEPT.PARTICIPANT.ID>"
```
Comment (znatty22) [inline docs/source/tutorial/extract.rst]: Why are all of the mapped concept col names enclosed in <>?

File: docs/source/tutorial/extract.rst [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+    "PID005", "", "", "", "1", "345", "TRUE", "TRUE", "34", "FALSE"
+    "PID006", "", "", "", "2", "34", "TRUE", "TRUE", "43545", "FALSE"
+    "PID007", "8", "9", "M", "1", "34", "TRUE", "FALSE", "5", "TRUE"
+    "PID008", "", "", "", "1", "43545", "TRUE", "TRUE", "52", "TRUE"
+    "PID009", "", "", "", "1", "5", "FALSE", "TRUE", "25", "TRUE"
+
+In order to ingest this data into the Kids First ecosystem, we need to:
+-----------------------------------------------------------------------
+
+* Convert the column headers into something standardized that the toolchain can
```
Comment (znatty22) [inline docs/source/tutorial/extract.rst]:
```suggestion
* Convert the column headers into a standardized set of values that the toolchain can understand. The values are defined in the Standard Concept Schema located  in kf_lib_data_ingest.etl.transform.standard_model.concept_schema.CONCEPT
```

File: docs/source/tutorial/extract.rst [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+``lambda x: int(x)`` could be replaced by just ``int``, since the two
+expressions are functionally equivalent (both single-argument functions that
+effectively strip the leading zeros).
+
+We could also have kept these IDs as they were and instead converted the
+mother/father IDs, but, in the absence of an overriding directive such as input
+from the investigators about their preferences, it doesn't really make a
+difference which way we choose.
+
+A value map operation with variable replacements
```
Comment (znatty22) [inline docs/source/tutorial/extract.rst]: I think we need a brief section before this next value map that describes the concept schema

```
Ok before moving on to the next operation, let's pause for a second and understand what CONCEPT.PARTICIPANT.ID is and where it came from ..

Understanding the Standard Concept Schema
-----------------------------------------------
At the most basic level, the Standard Concept Schema is just a set of standardized attributes used by the ingest library. Contextually, these attributes represent a standard set of things or concepts in the world and a standard set of attributes for those things or concepts. The ideas is to transform the source data to this standardized intermediate set of values and then transform from the intermediate values to a set of target values defined by the target database or service.

So for example the for a column named `Body site` in our source data, we need to tell the ingest library what this means by specifying which concept and attribute this column name translates to. In this case, the column would probably map to `CONCEPT.BIOSPECIMEN.ANATOMICAL_SITE`. 

In the case of our value map above, we are mapping the `participant` column to `CONCEPT.PARTICIPANT.ID`, something that means the unique identifier for the standard concept of participant.  
```

File: docs/source/tutorial/extract.rst [hunk 9]
```diff
@@ -0,0 +1,432 @@
...
+    value_map(
+        in_col="gender",
+        m={
+            "F": constants.GENDER.FEMALE,
+            "M": constants.GENDER.MALE
+        },
+        out_col=CONCEPT.PARTICIPANT.GENDER
+    )
+
+This says "Use the ``gender`` column as input (in_col=), replace everything
```
Comment (znatty22) [inline docs/source/tutorial/extract.rst]: Can we mention something about the fact that standard codes are defined in the constants module and you should always try to use them if possible?

File: docs/source/tutorial/extract.rst (hunk 8 above)
Comment (znatty22) [reply docs/source/tutorial/extract.rst]: Actually, should we have a separate section outside of extract for the standard concept schema?

File: docs/source/tutorial/project_setup.rst [hunk 10]
```diff
@@ -0,0 +1,31 @@
+.. _Tutorial-Project-Setup:
+
+=============
+Project Setup
+=============
+
+If you haven't done so already make sure to complete the installation before
+moving on.
+
+Create Your Ingest Project
```
[the remark this answers is not in the document]
Comment (fiendish) [reply docs/source/tutorial/project_setup.rst]: nah, just add the command quickly before this merges

File: docs/source/tutorial/extract.rst (hunk 6 above)
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: . vs | stylistic decision

File: docs/source/tutorial/extract.rst (hunk 8 above)
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: somewhere else

File: docs/source/tutorial/index.rst [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+a user how to run the ingest pipeline and view basic outputs. This tutorial
+will use a more complex study dataset and step through each ingest stage in
+much more depth.
+
+In this tutorial we will cover:
+
+1. How to configure complex extractions with multiple messy source data files
+2. How to write a transform function for the transform stage
+3. How to run the ingest pipeline from end to end
+4. How to debug steps 1 through 3 by:
```
[the remark this answers is not in the document]
Comment (fiendish) [reply docs/source/tutorial/index.rst]: 👌

File: docs/source/tutorial/project_setup.rst [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+==========================
+
+Create a new ingest project named ``My_Study`` by using the ingest library
```
[the remark this answers is not in the document]
Comment (fiendish) [reply docs/source/tutorial/project_setup.rst]: 👌

File: docs/source/tutorial/extract.rst [hunk 13]
```diff
@@ -0,0 +1,432 @@
...
+
+Supported protocol prefixes are:
+``file://``, ``s3://``, ``http://``, ``https://``
+
+Loading the data
+----------------
+
+.. code-block:: python
+
+    source_data_loading_parameters = {}
```
[the remark this answers is not in the document]
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: Hmm...yeah probably.

File: docs/source/tutorial/extract.rst (hunk 6 above)
Comment (znatty22) [reply docs/source/tutorial/extract.rst]: Oh I didn't even notice the delimiter 😄 , I was just talking about the <>

File: docs/source/tutorial/extract.rst (hunk 13 above)
[the remark this answers is not in the document]
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: 👌

File: docs/source/tutorial/extract.rst (hunk 2 above)
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: 👌

File: docs/source/tutorial/extract.rst (hunk 4 above)
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: 👌

File: docs/source/tutorial/extract.rst (hunk 5 above)
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: 👌

File: docs/source/tutorial/extract.rst (hunk 6 above)
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: I used < because I used . and not |

Comment (fiendish) [reply docs/source/tutorial/extract.rst]: it's arbitrary

File: docs/source/tutorial/extract.rst (hunk 7 above)
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: 👌

File: docs/source/tutorial/extract.rst (hunk 9 above)
Comment (fiendish) [reply docs/source/tutorial/extract.rst]: 👌

**Comment by fiendish** [comment]
rebasing to squash

**Pull request merged**